### PR TITLE
Changes to compile wps-sextante module with sextante libs 1.0

### PR DESCRIPTION
**Pull request by ahuarte47**
This PR update the wps-sextante module to compile with sextante libs 1.0

### CODE REVIEW

**Comment by aaime** [comment]
Checked, the pull request compiles. I did not check in the details of it, but as long as you're going to maintain the extension, that's none of my business either ;-) .
What is troublesome though is the copyright issue, many files do not have copyright headers, some have been written by people other than Alvaro (I checked, Alvaro has signed and sent the copyright assignement to OSGeo).

So, in order to accept the pull request we'd first need to clear out the copyright issues. Jody, could you help us here? What do we do with files that have a GPL v2.0 header and an author other than Alvaro, are they good and we just need to also add our own copyright header?
What about the files that are lacking a author reference and copyright altoghether? Alvaro, in this case I guess you could just write you're the author and add the copyright headers, but of course in case of any trouble with the file you're the one responsible.

**Comment by ahuarte47** [comment]
Hi Andrea, I updated the pull request. I added the copyright headers to all files, also the test unit is working.

Best Regards
Alvaro

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
```
Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: The headers here are wrong, the 2001-2013 Openplans line should be there only in files that were part of GeoServer before 2014. This one was not, it should only have the first line. And the same goes for other files in your pull request.

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import es.unex.sextante.core.Sextante;
+import es.unex.sextante.dataObjects.IRecord;
+import es.unex.sextante.dataObjects.IRecordsetIterator;
+import es.unex.sextante.dataObjects.RecordImpl;
+import es.unex.sextante.exceptions.IteratorException;
+
+/**
+ * 
+ * @author Cesar Martinez Izquierdo
```
Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: This is the author I'm concerned about, it's not the same person doing the pull request. Jody, what should we do in this case?

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/StreamOutputLayer.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+import es.unex.sextante.dataObjects.IFeatureIterator;
+import es.unex.sextante.dataObjects.IVectorLayer;
+import es.unex.sextante.dataObjects.vectorFilters.IVectorLayerFilter;
+import es.unex.sextante.outputs.IOutputChannel;
+
+/**
+ * A class to test the StreamOutputChannel class. It does nothing but streaming text descriptions of added features to a print
+ * stream
+ * 
+ * @author volaya
```
Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/StreamOutputLayer.java]: For this one we should ask Victor Olaya for authorization

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java (hunk 1 above)
Comment (ahuarte47) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Done, thanks Andrea!

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java (hunk 2 above)
Comment (jodygarnett) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Normally I would ask where the code came from (and if it is an open source project check the license it is made available under).

Alternatively we could reach out to the author and ask what the license is, or ask for explicit permission.  But yeah this is case where a contributor is introducing a dependency so we should check it out.

Normally we would check these things when a module has a code review (when it moves out of community).

Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Jody, there is a problem here, a module graduation has requirements, code review and IP check are not part of it, the module is supposed to be sane from IP point of view when it's committed to community already.

Comment (ahuarte47) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Hi, in last SIGTE conference ( http://www.sigte.udg.edu/jornadassiglibre/programa/jornadas/ ) celebrated in Girona (spain) we can speak with Victor Olaya and we inform him about this licesing requirement. In principle, we said no problems.

Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: The problem is that Victor is not Cesar. Do you have any way to contact Cesar instead?

Comment (ahuarte47) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: I think that it is his linkedin profile.
https://www.linkedin.com/pub/cesar-martinez-izquierdo/66/182/6ba

If you are agree, I can send a message

Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Yes, please try to contact him

Comment (ahuarte47) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTDBFIterator.java]: Ok, thanks Andrea!

**Comment by dispiste** [comment]
Hi, I am Cesar Martinez Izquierdo, I have been requested by Alvaro to join this discussion.
That code comes from the Geotools bindings for Sextante (quite old code). Its license is GPL 2.0.
Which is exactly the problem and how could I help to solve it?

**Comment by jodygarnett** [comment]
Afternoon Cesar:

Some code you have written has been provided as part of a pull-request by
Alvaro Huarte. As such we are contacting the original authors and making
sure everyone is fine with:

- the code being included in the GeoServer project.
- the code being contributed to the OSGeo foundation (see CONTRIBUTING.md
<https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md> ) for
details.

For reference GeoServer is GPL 2.0 (with an exception allowing us to use an
EPL library). We ask code to be contributed to OSGeo so we can keep our
options open about license change, or donating code to other projects (such
as GeoTools).


--
Jody Garnett

On 3 April 2015 at 10:10, Cesar Martinez Izquierdo <notifications@github.com

**Comment by dispiste** [comment]
Hi Jody, I am fine to contribute this code to Geoserver. I have sent a signed contributor agreement to info@osgeo.org. I hope this unblocks the pull request.

Greetings.

**Comment by ahuarte47** [comment]
Hi Cesar, thank you very much!

**Comment by jodygarnett** [comment]
Thank you Cesar that is exactly what needed :)

--
Jody Garnett

**Comment by aaime** [comment]
Merging then

**Comment by ahuarte47** [comment]
Thanks to all!

**Pull request merged**